### PR TITLE
HAAR-821: Added Delete endpoint

### PR DESCRIPTION
**Pull request by ravmoj**
Added Delete endpoint

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt [hunk 1]
```diff
@@ -209,6 +210,48 @@ class GroupsController(
...
+      )
+    ]
+  )
+  fun deleteGroup(
+    @Schema(description = "The group code of the group.", required = true)
+    @PathVariable
+    group: String
+  ) {
+    groupsService.deleteGroup(group)
+  }
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt:254]: can just do = groupsService.deleteGrou(group)  - curly brackets not needed.
Not a show-stopper though

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt [hunk 2]
```diff
@@ -414,4 +415,57 @@ class GroupsControllerIntTest : IntegrationTestBase() {
...
+          """.trimIndent()
+        )
+    }
+    @Test
+    fun `Delete Child Group details endpoint not accessible without valid token`() {
+      webTestClient.delete().uri("/groups/GC_DEL_1")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+  }
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt:470]: Should there be a test of no group exists? - check for 404 / not found?

**Review by karenmillermoj — APPROVED** [review]
lgtm

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt (hunk 1 above)
Comment (ravmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsController.kt]: Done

File: src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt [hunk 3]
```diff
@@ -578,4 +578,80 @@ class GroupsControllerIntTest : IntegrationTestBase() {
...
+    }
+
+    @Test
+    fun `Delete Group - has child groups`() {
+      externalUsersApiMockServer.stubDeleteGroupsConflict()
+      webTestClient.delete().uri("/groups/GC_DEL_3")
+        .headers(setAuthorisation("ITAG_USER_ADM", listOf("ROLE_MAINTAIN_OAUTH_USERS")))
+        .exchange()
+        .expectStatus().isEqualTo(HttpStatus.CONFLICT)
+        .expectHeader().contentType(MediaType.APPLICATION_JSON)
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/manageusersapi/resource/GroupsControllerIntTest.kt:600]: Minor: Can import CONFLICT and APPLICATION_JSON

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**